MapboxNavigation/NavigationMapView.swift

```diff
@@ -142,20 +149,25 @@ open class NavigationMapView: MGLMapView {
             addAnnotation(destination)
         }
         
-        let source = navigationMapDelegate?.navigationMapView?(self, shapeFor: routeProgress.remainingWaypoints) ?? shape(for: Array(routeProgress.remainingWaypoints.dropLast()))
+        let source = navigationMapDelegate?.navigationMapView?(self, shapesFor: routeProgress.remainingWaypoints) ?? shapes(for: Array(routeProgress.remainingWaypoints.dropLast()))
         
         if let s = style.source(withIdentifier: waypointSourceIdentifier) as? MGLShapeSource {
             s.shape = source
         } else {
             let sourceShape = MGLShapeSource(identifier: waypointSourceIdentifier, shape: source, options: nil)
             style.addSource(sourceShape)
             
-            let circles = navigationMapDelegate?.navigationMapView?(self, routeWaypointCircleStyleLayerWithIdentifier: waypointCircleIdentifier, source: sourceShape) ?? routeWaypointCircleStyleLayer(identifier: waypointCircleIdentifier, source: sourceShape)
+            let circles = navigationMapDelegate?.navigationMapView?(self, waypointStyleLayerWithIdentifier: waypointCircleIdentifier, source: sourceShape) ?? routeWaypointCircleStyleLayer(identifier: waypointCircleIdentifier, source: sourceShape)
             
-            let symbols = navigationMapDelegate?.navigationMapView?(self, routeWaypointSymbolStyleLayerWithIdentifier: waypointSymbolIdentifier, source: sourceShape) ?? routeWaypointSymbolStyleLayer(identifier: waypointSymbolIdentifier, source: sourceShape)
+            let symbols = navigationMapDelegate?.navigationMapView?(self, waypointSymbolStyleLayerWithIdentifier: waypointSymbolIdentifier, source: sourceShape) ?? routeWaypointSymbolStyleLayer(identifier: waypointSymbolIdentifier, source: sourceShape)
```

Comment:
If the developer wants the waypoint to be marked by a circle with no letter, they’d return `nil` for `navigationMapView(_:waypointSymbolStyleLayerWithIdentifier:source:)`, but then we’d go ahead and add a letter anyways.